ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -628,9 +632,14 @@ runThreadNetwork ThreadNetworkArgs
               -> traceWith addTracer (p, bno)
           _   -> pure ()
 
+    decorateId :: CoreNodeId -> Tracer m String -> Tracer m String
+    decorateId (CoreNodeId cid) = contramap $ \s ->
+        show cid <> " | " <> s
```

Comment:
Doesn't this give an "unused binding" warning (CI should run with `-Werror`, so we'll see)? (Same for the `coreNodeId` argument. I think it's useful to leave them in for debugging purposes. Maybe prefix them with an underscore to suppress the warning and a comment to `decorateId` on how to use it for debugging?